src/ts/plugins/server/GitlabHostingConnector.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Silex website builder, free/libre no-code tool for makers.
+ * Copyright (c) 2023 lexoyo and Silex Labs foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Gitlab hosting connector
+ * @fileoverview Gitlab hosting connector for Silex, connect to the user's Gitlab account to host websites with .gitlab-ci.yml file template for plain html by default
+ * @see https://docs.gitlab.com/ee/api/oauth2.html
+ * @see https://docs.gitlab.com/ee/user/project/pages/getting_started/pages_ci_cd_template.html
+ */
+
+import GitlabConnector, { GitlabOptions, GitlabToken } from './GitlabConnector'
+import { HostingConnector, ConnectorFile } from '../../server/connectors/connectors'
+import { ConnectorType, WebsiteId, JobData, JobStatus, PublicationJobData } from '../../types'
+import { JobManager } from '../../server/jobs'
+import { join } from 'path'
+import { ServerConfig } from '../../server/config'
+
+//type GitlabSession = ConnectorSession
+//type GitlabOption = GitlabOptions
+type GitlabSession = Record<string, GitlabToken>
+
+export default class GitlabHostingConnector extends GitlabConnector implements HostingConnector/*<GitlabSession>*/ {
+  connectorId = 'gitlab_hosting'
```

Comment:
the session problems you have probably come from here

i see [here in GitlabConnector::getSessionToken](https://github.com/silexlabs/Silex/blob/dev/src/ts/plugins/server/GitlabConnector.ts#L397) that it uses `connectorId`

So your problem comes from there: the session is in `{ gitlab: { token } }` but in your instance it will look for `{ gitlab_hosting: { token } }` 

I believe you can should remove this line and it will work as expected (it will use `gitlab` as connectorId)

Reply:
yes, working, thx